R/health-plots.R

```diff
@@ -68,29 +75,49 @@ riskfactors_diff <- function(riskft_diff, data_mean_weighted_rf_wide) {
                     sodium = "Sodium")
 
   y_value <- switch(riskft_diff,
-                    bmi = "diff_bmi",
-                    ei = "diff_ei",
-                    obesity = "diff_obesity",
-                    sodium = "diff_sodium")
+                    bmi = "diff_bmi_mean",
+                    ei = "diff_ei_mean",
+                    obesity = "diff_obesity_mean",
+                    sodium = "diff_sodium_mean")
+
+  y_min <- switch(riskft_diff,
+                  bmi = "diff_bmi_min",
+                  ei = "diff_ei_min",
+                  obesity = "diff_obesity_min",
+                  sodium = "diff_sodium_min")
+
+  y_max <- switch(riskft_diff,
+                  bmi = "diff_bmi_max",
+                  ei = "diff_ei_max",
+                  obesity = "diff_obesity_max",
+                  sodium = "diff_sodium_max")
 
   plot_title <- switch(riskft_diff,
-                       bmi = "Reduction in BMI under intervention",
-                       ei = "Reduction in energy intake (kcal) under intervention",
-                       obesity = "Reduction in obesity prevalence under intervention",
-                       sodium = "Reduction in sodium (mg) under intervention")
+                       bmi = "Reduction in BMI by income class",
+                       ei = "Reduction in energy intake (kcal) by income class",
+                       obesity = "Reduction in obesity prevalence by income class",
+                       sodium = "Reduction in sodium (mg) by income class")
 
-  ggplot2::ggplot(data = data_mean_weighted_rf_wide,
-                  ggplot2::aes(x = data_mean_weighted_rf_wide$timediff,
-                               y = get(y_value))) +
-    ggplot2::geom_line(colour = "blue", linewidth = 1) +
+  ggplot2::ggplot(data = data_weighted_rf_wide_collapse,
+                  ggplot2::aes(x = data_weighted_rf_wide_collapse$time,
+                               y = get(y_value),
+                               colour = data_weighted_rf_wide_collapse$income)) +
+    ggplot2::geom_line(linewidth = 1) +
+    ggplot2::geom_ribbon(ggplot2::aes(ymin = y_min,
+                                      ymax = y_max),
+                         alpha = 0.2) +
     ggplot2::ggtitle(plot_title) +
     ggplot2::xlab("Year") +
     ggplot2::ylab(y_label) +
     ggplot2::scale_x_continuous(limits = c(-3, 32),
                        breaks = c(-3, 2, 7, 12, 17, 22, 27, 32),
                        labels = c(2020, 2025, 2030, 2035, 2040, 2045, 2050, 2055)) +
+    ggplot2::scale_y_continuous(limits = scale_y_continuous_limits,
+                       breaks = scale_y_continuous_breaks,
+                       labels = scale_y_continuous_labels) +
     ggplot2::labs(alt = "A line plot showing the reduction in a specified risk factor under intervention over time") +
-    hgps_theme()
+    hgps_theme() +
+    ggplot2::theme(legend.position.inside = c(0.85,0.22))
```

Comment:
I am unsure whether it's a good idea to hard-code this legend position, as well as the scale of x axis.
Because the position was adjusted based on the plot. It can be on the left or right. Regarding the scale of x axis, for future projects, the simulation years might be different. Maybe we just keep it like this for now and bear this in mind.